packages/core-db/src/app/queue/sequential-processing-data-service.ts

```diff
@@ -0,0 +1,173 @@
+/* External Imports */
+import { getLogger, Logger, logError } from '@eth-optimism/core-utils'
+
+/* Internal Imports */
+import {
+  SequentialProcessingDataService,
+  SequentialProcessingItem,
+} from '../../types/queue'
+import { RDB, Row } from '../../types/db'
+
+const log: Logger = getLogger('sequential-processing-data-service')
+
+export class DefaultSequentialProcessingDataService
+  implements SequentialProcessingDataService {
+  constructor(private readonly rdb: RDB) {}
+
+  /**
+   * @inheritDoc
+   */
+  public async fetchItem(
+    index: number,
+    sequenceKey: string
+  ): Promise<SequentialProcessingItem> {
+    const res: Row[] = await this.rdb.select(
+      `SELECT data_to_process, processed
```

Comment:
I recall you having a comment about sql injection attacks, is there a template engine that protects against them rather than backticks? The last time I was doing SQL in JS, I was using SQLite and there was a builtin method that did the templating for you and made sure it was safe.

Reply:
Yep, there are ways to protect as well as prepared statements, but in this case we’ll only be adding data that we specifically create (Sequence keys) or stringified block data & block numbers both of which should fail parsing if not the correct type. Let me know if you see anything that could be problematic, though!

Reply:
Thinking about it more, it’s probably somewhat low effort to add SQL injection protection. Will create a ticket for it. The stuff that is wrapped in quotes shouldn’t be an issue but it could be a good thing to not have to worry about. Stuff that is assumed to be a number but is in JS and technically may not be could be problematic if we’re not scrutinizing every PR through that lens moving forward.

Reply:
Created: https://optimists.atlassian.net/browse/YAS-631